src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/CaseCreationCallbackHandlerTest.java

```diff
@@ -0,0 +1,122 @@
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.handler.ExceptionRecordEventHandler;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.handler.model.CaseCreationRequest;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.handler.model.CaseCreationResult;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.handler.model.ExceptionRecord;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.services.exception.CallbackProcessingException;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.willThrow;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+public class CaseCreationCallbackHandlerTest {
+
+    @Mock
+    private ExceptionRecordEventHandler exceptionRecordEventHandler;
+
+    private CaseCreationCallbackHandler callbackHandler;
+
+    @BeforeEach
+    public void setUp() {
+        callbackHandler = new CaseCreationCallbackHandler(exceptionRecordEventHandler);
+    }
+
+    @Test
+    public void should_return_result_from_exception_record_event_handler() {
+        // given
+        CaseCreationResult expectedResult =
+            new CaseCreationResult("case-id-123", emptyList(), singletonList("warning 1"));
+
+        given(exceptionRecordEventHandler.handle(any())).willReturn(expectedResult);
+
+        String eventId = "eventId1";
+        String idamToken = "idamToken1";
+        String idamUserId = "idamUser1";
+
+        CaseDetails exceptionRecordDetails = sampleValidExceptionRecordDetails();
+
+        // when
+        CaseCreationResult actualResult = callbackHandler.handleCaseCreationCallback(
+            exceptionRecordDetails,
+            eventId,
+            idamToken,
+            idamUserId,
+            true
+        );
+
+        // then
+        assertThat(actualResult).isSameAs(expectedResult);
+
+        verifyEventHandlerHasBeenCalled(exceptionRecordDetails, eventId, idamToken, idamUserId);
+    }
+
+    @Test
+    public void should_throw_exception_when_exception_record_event_handler_fails() {
+        Exception exceptionToThrow = new CallbackProcessingException("test exception", null);
+        willThrow(exceptionToThrow).given(exceptionRecordEventHandler).handle(any());
+
+        assertThatThrownBy(
+            () -> callbackHandler.handleCaseCreationCallback(
+                sampleValidExceptionRecordDetails(),
+                "eventId1",
+                "idamToken1",
+                "idamUserId1",
+                true
+            )
+        ).isSameAs(exceptionToThrow);
+    }
+
+    private void verifyEventHandlerHasBeenCalled(
```

Comment:
we are doing more than what the method name says. I suggest renaming the method. otherwise, looks good

Reply:
`verifyEventHandlerHasBeenCalledWithRightData`?

Reply:
how about `verifyEventHandlerHasBeenCalledWithExpectedData`?

Reply:
maybe `...correctData`?